spec/queries/books/best_seller_query_spec.rb

```diff
@@ -0,0 +1,77 @@
+require 'rails_helper'
+
+describe Books::BestSellersQuery do
+  def book_find(arr, title)
+    arr.each do |item|
+      return item if item.title == title
+    end
+  end
+
+  def entity_save(arr)
+    arr.each do |item|
+      item.save
+    end
+  end
+
+  describe '#call' do
+    let(:orders) { create_list(:order, 4, state: 'in_queue') }
+    let(:books) { create_list(:book, 40) }
+    let(:categories) { create_list(:category, 4) }
+
+    context 'when there are bestsellers in each of the four categories' do
```

Comment:
please refactor this test at all
it's hard to understand what is happening
Ideally it would be great if you create all data you need with `let`
and your test should looks like
```ruby

it 'returns four bestsellers' do
  expect(Books::BestSellersQuery.new.call).to include(expected_books)
end
```